arbor/fvm_layout.cpp

```diff
@@ -315,11 +313,18 @@ ARB_ARBOR_API fvm_cv_discretization fvm_cv_discretize(const cable_cell& cell, co
                 id_map.insert(k, 1.0/v.value);
             }
         }
-        // Fetch defaults, either global or per cell
-        auto gd = *(value_by_key(dflt.ion_data,        ion).value().diffusivity
-                  | value_by_key(global_dflt.ion_data, ion).value().diffusivity);
-        if (gd <= 0.0) {
-            throw cable_cell_error{util::pprintf("Illegal global diffusivity '{}' for ion '{}'.", gd, ion)};
+        arb_value_type def = 0.0;
+        if (auto data = value_by_key(global_dflt.ion_data, ion);
+            data && data->diffusivity) {
```

Comment:
the if statement with initializer is is a c++17 feature I was not aware of! cool

Reply:
It's extremely handy.